src/Channels/ReferenceCountedBuffer.cs

```diff
@@ -28,19 +25,16 @@ public IBuffer Preserve(int offset, int count, out int newStart, out int newEnd)
 
         public void Dispose()
         {
-            lock (_lockObj)
```

Comment:
In theory, this buffer can be leasable with the wrong reference count after DisposeBuffer is called and before the count is reset to 1.